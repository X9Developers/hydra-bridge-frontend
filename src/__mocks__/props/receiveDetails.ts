import { IconKeys } from "../../common/components/Atoms/Icons/Icon";
import { ReceiveDetailsAccordionProps } from "../../common/components/Atoms/ReceiveDetails/ReceiveDetailsAccordion";

export const mockReceiveDetailsProps: ReceiveDetailsAccordionProps = {
  iconKey: "hopChain" as IconKeys,
  chainName: "Hop",
  gasFees: 0.39979797399225586,
  serviceTime: 30,
  transactionFees: "0.00001",
  slippage: "0.00001",
  amountOut: "0.00001",
  symbol: "ETH",
};
