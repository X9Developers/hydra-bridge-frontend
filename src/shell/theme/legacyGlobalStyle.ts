import { createGlobalStyle } from "styled-components";

export const LegacyGlobalStyle = createGlobalStyle`
body, html {
  height: 100%;
}

body {
  background-color: #fae3f8;
  font-family: 'Roboto', sans-serif;
}

/* Chrome, Safari, Edge, Opera */
input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type=number] {
  -moz-appearance: textfield;
}

`;
