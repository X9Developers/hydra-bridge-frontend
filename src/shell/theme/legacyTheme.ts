import { DefaultTheme, LegacyTheme } from "styled-components";

declare module "styled-components" {
  export interface LegacyTheme extends DefaultTheme {
    primaryColor: string;
    secondaryColor: string;

    heading: {
      xxl: string;
      xl: string;
      lg: string;
      md: string;
      sm: string;
      xs: string;
    };

    paragraph: {
      xxl: string;
      xl: string;
      lg: string;
      md: string;
      sm: string;
      xs: string;
    };

    borderRadius: {
      xs: string;
      sm: string;
      md: string;
      lg: string;
    };
    backgroundColor: string;
    lightBackgroundColorHover: string;
    inputBackgroundColor: string;
    inputBorderColor: string;
    lightBackgroundColor: string;
    greyColor: string;
    buttonDefaultColor: string;
    greenColor: string;
    blueColor: string;
    redColor: string;
  }
}

export const legacyTheme: LegacyTheme = {
  primaryColor: "#ffffff",
  secondaryColor: "#121212",

  heading: {
    xxl: "5.8rem",
    xl: "4.8rem",
    lg: "3.8rem",
    md: "2.8rem",
    sm: "2.4rem",
    xs: "1.9rem",
  },

  paragraph: {
    xxl: "2.2rem",
    xl: "1.8rem",
    lg: "1.6rem",
    md: "1.4rem",
    sm: "1.2rem",
    xs: "1rem",
  },

  borderRadius: {
    xs: ".3rem",
    sm: ".6rem",
    md: "1rem",
    lg: "2rem",
  },
  backgroundColor: "#1e1e1e",
  lightBackgroundColorHover: "#171717",
  inputBackgroundColor: "rgba(255, 255, 255, 0.05)",
  inputBorderColor: "rgba(255, 255, 255, 0.3)",
  lightBackgroundColor: "#2d2d2d",
  greyColor: "rgb(226, 226, 229)",
  buttonDefaultColor: "rgba(0,0,0,0.2)",
  greenColor: "#37b514",
  blueColor: "#1976d2",
  redColor: "#d32f2f",
};
