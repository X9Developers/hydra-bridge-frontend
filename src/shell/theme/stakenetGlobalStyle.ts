import { createGlobalStyle } from "styled-components";

export const StakenetGlobalStyle = createGlobalStyle`
  body, html {
    height: 100%;
    /* converts 1 rem to 10px instead of 16px */
    font-size: 62.5% !important;
  }

  body {
    /* brand darkest blue */
    background-color: #0D1328;
    font-family: 'Rubik', sans-serif;
    line-height: 1.1;
  }

  /* Chrome, Safari, Edge, Opera */
  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  /* Firefox */
  input[type=number] {
    -moz-appearance: textfield;
  }
`;
