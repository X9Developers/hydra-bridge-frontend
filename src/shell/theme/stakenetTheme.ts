import { DefaultTheme } from "styled-components";

export interface IStakenetTheme extends DefaultTheme {
  colors: {
    blue: {
      darkest: string;
      darker: string;
      dark: string;
      medium: string;
      light: string;
      lighter: string;
      lightest: string;
      "medium-dark": string;
    };
    white: string;
    black: string;
    gray: {
      darker: string;
      dark: string;
      "medium-dark": string;
    };
  };

  heading: {
    xxl: string;
    xl: string;
    lg: string;
    md: string;
    sm: string;
    xs: string;
  };

  paragraph: {
    xl: string;
    lg: string;
    md: string;
    sm: string;
    xs: string;
  };

  margin: {
    lg: string;
    md: string;
    sm: string;
    none: string;
  };

  fontWeight: {
    bold: number;
    semibold: number;
    medium: number;
    normal: number;
    light: number;
  };

  borderRadius: {
    lg: string;
    md: string;
    sm: string;
    xs: string;
  };

  maxWidth: {
    full: string;
    "7xl": string;
    "6xl": string;
    "5xl": string;
    "4xl": string;
    "3xl": string;
    "2xl": string;
    xl: string;
    lg: string;
    md: string;
    sm: string;
    xs: string;
  };
}

export const stakenetTheme: IStakenetTheme = {
  colors: {
    blue: {
      darkest: "#0D1328",
      darker: "#192038",
      dark: "#0D132830",
      "medium-dark": "#3F476D",
      medium: "#1254DD",
      light: "#1356DE",
      lighter: "#7377A5",
      lightest: "#1D96EC",
    },
    white: "#FFFFFF",
    black: "#000000",
    gray: {
      darker: "#7377A514",
      dark: "#7377A529",
      "medium-dark": "#7377A53D",
    },
  },

  heading: {
    xxl: "5.8rem",
    xl: "4.8rem",
    lg: "3.8rem",
    md: "2.8rem",
    sm: "2.4rem",
    xs: "1.9rem",
  },

  paragraph: {
    xl: "2rem",
    lg: "1.8rem",
    md: "1.6rem",
    sm: "1.4rem",
    xs: "1.3rem",
  },

  margin: {
    lg: "3.2rem",
    md: "1.2rem",
    sm: ".6rem",
    none: "0rem",
  },

  fontWeight: {
    bold: 700,
    semibold: 600,
    medium: 500,
    normal: 400,
    light: 300,
  },

  borderRadius: {
    lg: ".8rem",
    md: ".6rem",
    sm: ".4rem",
    xs: ".2rem",
  },

  maxWidth: {
    full: "100%",
    "7xl": "80rem",
    "6xl": "72rem",
    "5xl": "64rem",
    "4xl": "56rem",
    "3xl": "48rem",
    "2xl": "42rem",
    xl: "36rem",
    lg: "32rem",
    md: "28rem",
    sm: "24rem",
    xs: "20rem",
  },
};
