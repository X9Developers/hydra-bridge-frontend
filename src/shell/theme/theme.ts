import { DefaultTheme } from 'styled-components';

declare module 'styled-components' {
  export interface DefaultTheme {
    primaryColor: string;
    secondaryColor: string;

    heading: {
      xxl: string;
      xl: string;
      lg: string;
      md: string;
      sm: string;
      xs: string;
    };

    paragraph: {
      xxl: string;
      xl: string;
      lg: string;
      md: string;
      sm: string;
      xs: string;
    };

    borderRadius: {
      xs: string;
      sm: string;
      md: string;
      lg: string;
    };
    backgroundColor: string;
    lightBackgroundColorHover: string;
    inputBackgroundColor: string;
    inputBorderColor: string;
    lightBackgroundColor: string;
    greyColor: string;
    buttonDefaultColor: string;
    greenColor: string;
    blueColor: string;
    redColor: string;
  }
}

export const defaultTheme: DefaultTheme = {
  primaryColor: '#ffffff',
  secondaryColor: '#121212',

  heading: {
    xxl: '58px',
    xl: '48px',
    lg: '38px',
    md: '28px',
    sm: '24px',
    xs: '19px',
  },

  paragraph: {
    xxl: '22px',
    xl: '18px',
    lg: '16px',
    md: '14px',
    sm: '12px',
    xs: '10px',
  },

  borderRadius: {
    xs: '3px',
    sm: '6px',
    md: '10px',
    lg: '20px',
  },
  backgroundColor: '#1e1e1e',
  lightBackgroundColorHover: '#171717',
  inputBackgroundColor: 'rgba(255, 255, 255, 0.05)',
  inputBorderColor: 'rgba(255, 255, 255, 0.3)',
  lightBackgroundColor: '#2d2d2d',
  greyColor: 'rgb(226, 226, 229)',
  buttonDefaultColor: 'rgba(0,0,0,0.2)',
  greenColor: '#37b514',
  blueColor: '#1976d2',
  redColor: '#d32f2f',
};
