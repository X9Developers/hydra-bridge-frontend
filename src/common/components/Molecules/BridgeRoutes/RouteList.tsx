import React, { useMemo } from "react";
import { useTranslation } from "react-i18next";

import RouteItem, { RouteItemContainerCard } from "./RouteItem";
import RouteItemFees from "./RouteItemFees";

import { RouteDto } from "../../../dtos";
import { getBridgeIcon, getCoinIcon } from "../../../../helpers/icons";
import { stakenetTheme as theme } from "../../../../shell/theme/stakenetTheme";
import { IInlineStyles } from "../../../commonTypes";

const styles: IInlineStyles = {
  routeItemWrapper: { width: "100%", marginBottom: theme.margin.md },
  routeItemContainerCard: { width: "100%", marginBottom: theme.margin.md },
  errorShowingRoutes: {
    fontSize: theme.paragraph.md,
    color: theme.colors.red,
    margin: "auto",
    textAlign: "center" as const,
  },
};

export type RouteListProps = {
  routes: RouteDto[];
  selectedRouteId?: number;
  onRouteSelect: (routeId: number) => void;
};

const RouteList = ({
  routes,
  selectedRouteId,
  onRouteSelect,
}: RouteListProps) => {
  const { t } = useTranslation();

  const renderRouteItems = useMemo(() => {
    return routes.map((route: RouteDto) => {
      try {
        const {
          id,
          transactionCoastUsd,
          bridgeRoute: {
            amountIn,
            amountOut,
            bridgeName,
            fromAsset: { symbol },
            bridgeInfo: { displayName, serviceTime },
          },
        } = route;
        const coinSymbol = getCoinIcon(symbol);
        const bridgeSymbol = getBridgeIcon(bridgeName);

        return (
          <div key={id} id={`route-${id}`} style={styles.routeItemWrapper}>
            <RouteItem
              coinSymbol={coinSymbol}
              bridgeSymbol={bridgeSymbol}
              amountIn={amountIn}
              amountOut={amountOut}
              bridgeDisplayName={displayName}
              routeId={id}
              isSelected={selectedRouteId === id}
              onRouteSelect={onRouteSelect}
            >
              <RouteItemFees
                transactionCostInUsd={transactionCoastUsd}
                serviceTime={serviceTime}
              />
            </RouteItem>
          </div>
        );
      } catch (err) {
        console.warn(
          "Couldn't access required values from routes to display it",
          route
        );
        return (
          <RouteItemContainerCard
            key={route?.id}
            isSelected={false}
            hasError={true}
            style={styles.routeItemContainerCard}
          >
            <p style={styles.errorShowingRoutes}>
              {t("errors.showing-routes")}
            </p>
          </RouteItemContainerCard>
        );
      }
    });
  }, [routes, selectedRouteId]);
  return <>{renderRouteItems}</>;
};
export default React.memo(RouteList);
