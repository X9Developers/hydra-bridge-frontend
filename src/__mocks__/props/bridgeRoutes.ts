import { RouteItemProps } from "../../common/components/Molecules/BridgeRoutes/RouteItem";
import { RouteListProps } from "../../common/components/Molecules/BridgeRoutes/RouteList";
import { BridgeRoutesProps } from "../../common/components/Molecules/BridgeRoutes/BridgeRoutes";
import QuoteResponseDtoMock from "../fromResponse/QuoteResponseDto.json";

export const mockRouteItemBaseProps: RouteItemProps = {
  amountIn: "0.0001",
  amountOut: "0.0001",
  coinSymbol: "ethCoin",
  bridgeSymbol: "polygonChain",
  bridgeDisplayName: "Polygon",
  routeId: 2,
  isSelected: false,
  onRouteSelect: (id: number) => alert(`on route select id: ${id}`),
};

export const mockRouteListBaseProps: RouteListProps = {
  routes: QuoteResponseDtoMock.routes,
  selectedRouteId: QuoteResponseDtoMock.routes[0].id,
  onRouteSelect: (id: number) => alert(`on route select id: ${id}`),
};

export const mockBridgeRoutesBaseProps: BridgeRoutesProps = {
  routes: QuoteResponseDtoMock.routes,
  selectedRouteId: QuoteResponseDtoMock.routes[0].id,
  inProgress: false,
  onRouteSelect: (id: number) => alert(`on route select id: ${id}`),
};
