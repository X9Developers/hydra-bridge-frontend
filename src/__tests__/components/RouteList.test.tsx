import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import _ from "lodash";

import RouteList, {
  RouteListProps,
} from "../../common/components/Molecules/BridgeRoutes/RouteList";

import { mockRouteListBaseProps } from "../../__mocks__/props/bridgeRoutes";
import { stakenetTheme as theme } from "../../shell/theme/stakenetTheme";
import QuoteResponseDtoMock from "../../__mocks__/fromResponse/QuoteResponseDto.json";

describe("The Bridge route list", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render with routes", async () => {
    const props: RouteListProps = mockRouteListBaseProps;

    const { asFragment } = render(<RouteList {...props} />);
    expect(asFragment()).toMatchSnapshot();
    render(<RouteList {...props} />);
  });

  it("should render the routes, when a error occurs", async () => {
    const quoteResponseWithOneRouteError = _.cloneDeep(QuoteResponseDtoMock);
    // @ts-ignore delete
    delete quoteResponseWithOneRouteError.routes[1].bridgeRoute;
    const props: RouteListProps = {
      ...mockRouteListBaseProps,
      routes: quoteResponseWithOneRouteError.routes,
    };
    const { container, asFragment } = render(<RouteList {...props} />);

    const routeWithoutError = container.querySelector<HTMLDivElement>(
      `#route-${props.routes[0].id}`
    );
    const routeWithError = screen.getByText(/errors.showing-route/);

    expect(asFragment()).toMatchSnapshot();
    expect(container.childElementCount).toBe(props.routes.length);
    expect(routeWithoutError).toBeInTheDocument();
    // @ts-ignore possible null
    expect(routeWithoutError.childElementCount).toBe(1);
    expect(routeWithError).toBeInTheDocument();
    expect(routeWithError).toHaveStyle(`color: ${theme.colors.red}`);
  });

  it("should render the routes, when all of them have errors", async () => {
    const quoteResponseWithAllRouteError = _.cloneDeep(QuoteResponseDtoMock);
    // @ts-ignore delete
    delete quoteResponseWithAllRouteError.routes[0].bridgeRoute;
    // @ts-ignore delete
    delete quoteResponseWithAllRouteError.routes[1].bridgeRoute;

    const props: RouteListProps = {
      ...mockRouteListBaseProps,
      routes: quoteResponseWithAllRouteError.routes,
    };
    const { container, asFragment } = render(<RouteList {...props} />);

    const routesWithError = screen.getAllByText(/errors.showing-route/);

    expect(asFragment()).toMatchSnapshot();
    expect(container.childElementCount).toBe(props.routes.length);
    routesWithError.forEach((route) => {
      expect(route).toBeInTheDocument();
      expect(route).toHaveStyle(`color: ${theme.colors.red}`);
    });
  });
});
