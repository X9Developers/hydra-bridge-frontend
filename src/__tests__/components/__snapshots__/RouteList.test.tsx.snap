// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Bridge route list should render the routes, when a error occurs 1`] = `
<DocumentFragment>
  <div
    id="route-8"
    style="width: 100%; margin-bottom: 1.2rem;"
  >
    <div
      aria-expanded="false"
      aria-label="select route"
      class="sc-iqseJM ibXHyJ sc-crHmcD gYEbqi"
      role="button"
      tabindex="0"
    >
      <div
        class="sc-bdvvtL sc-jRQBWg fGXmcv gzSVnD sc-egiyK eqEoBy"
      >
        <div
          class="sc-gKclnd konHGI"
        >
          <div
            class="sc-iCfMLu gYGgwA"
          >
            <svg
              class="amount__icon"
              height="3.2rem"
              name="usdcCoin"
              width="2.8rem"
            />
            <span
              class="amount__number"
            >
              1
            </span>
          </div>
          <div
            class="sc-furwcr fhgROL"
          >
            <svg
              class="arrow__icon"
              height="1.8rem"
              name="doubleArrowRight"
              width="2.1rem"
            />
          </div>
          <div
            class="sc-pVTFL joWhdh"
          >
            <svg
              class="chain__icon chain__icon--sm"
              height="2.6rem"
              name="hopChain"
              width="2.6rem"
            />
            <p
              class="chain__name"
            >
              Hop Goerli
            </p>
            <div
              class="chain__group"
            >
              <span>
                <svg
                  class="network__icon"
                  height="2.6rem"
                  name="hopChain"
                  width="2.6rem"
                />
              </span>
              <span>
                Hop Goerli
              </span>
            </div>
          </div>
          <div
            class="sc-furwcr fhgROL"
          >
            <svg
              class="arrow__icon"
              height="1.8rem"
              name="doubleArrowRight"
              width="2.1rem"
            />
          </div>
          <div
            class="sc-iCfMLu fRazBy"
          >
            <svg
              class="amount__icon"
              height="3.2rem"
              name="usdcCoin"
              width="2.8rem"
            />
            <span
              class="amount__number"
            >
              1.0
            </span>
          </div>
        </div>
        <div
          class="sc-bdvvtL sc-jRQBWg fGXmcv fmljMh"
        >
          <p
            class="sc-bqiRlB kMHECB"
            data-testid="route-fees"
          >
            gas-fee : ~$0.61 | ~5 min
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-bdvvtL sc-jRQBWg fGXmcv gzSVnD sc-egiyK jMIVTt"
    style="width: 100%; margin-bottom: 1.2rem;"
  >
    <p
      style="font-size: 1.6rem; color: rgb(191, 44, 68); margin: auto; text-align: center;"
    >
      errors.showing-routes
    </p>
  </div>
</DocumentFragment>
`;

exports[`The Bridge route list should render the routes, when all of them have errors 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL sc-jRQBWg fGXmcv gzSVnD sc-egiyK jMIVTt"
    style="width: 100%; margin-bottom: 1.2rem;"
  >
    <p
      style="font-size: 1.6rem; color: rgb(191, 44, 68); margin: auto; text-align: center;"
    >
      errors.showing-routes
    </p>
  </div>
  <div
    class="sc-bdvvtL sc-jRQBWg fGXmcv gzSVnD sc-egiyK jMIVTt"
    style="width: 100%; margin-bottom: 1.2rem;"
  >
    <p
      style="font-size: 1.6rem; color: rgb(191, 44, 68); margin: auto; text-align: center;"
    >
      errors.showing-routes
    </p>
  </div>
</DocumentFragment>
`;

exports[`The Bridge route list should render with routes 1`] = `
<DocumentFragment>
  <div
    id="route-8"
    style="width: 100%; margin-bottom: 1.2rem;"
  >
    <div
      aria-expanded="false"
      aria-label="select route"
      class="sc-iqseJM ibXHyJ sc-crHmcD gYEbqi"
      role="button"
      tabindex="0"
    >
      <div
        class="sc-bdvvtL sc-jRQBWg fGXmcv gzSVnD sc-egiyK eqEoBy"
      >
        <div
          class="sc-gKclnd konHGI"
        >
          <div
            class="sc-iCfMLu gYGgwA"
          >
            <svg
              class="amount__icon"
              height="3.2rem"
              name="usdcCoin"
              width="2.8rem"
            />
            <span
              class="amount__number"
            >
              1
            </span>
          </div>
          <div
            class="sc-furwcr fhgROL"
          >
            <svg
              class="arrow__icon"
              height="1.8rem"
              name="doubleArrowRight"
              width="2.1rem"
            />
          </div>
          <div
            class="sc-pVTFL joWhdh"
          >
            <svg
              class="chain__icon chain__icon--sm"
              height="2.6rem"
              name="hopChain"
              width="2.6rem"
            />
            <p
              class="chain__name"
            >
              Hop Goerli
            </p>
            <div
              class="chain__group"
            >
              <span>
                <svg
                  class="network__icon"
                  height="2.6rem"
                  name="hopChain"
                  width="2.6rem"
                />
              </span>
              <span>
                Hop Goerli
              </span>
            </div>
          </div>
          <div
            class="sc-furwcr fhgROL"
          >
            <svg
              class="arrow__icon"
              height="1.8rem"
              name="doubleArrowRight"
              width="2.1rem"
            />
          </div>
          <div
            class="sc-iCfMLu fRazBy"
          >
            <svg
              class="amount__icon"
              height="3.2rem"
              name="usdcCoin"
              width="2.8rem"
            />
            <span
              class="amount__number"
            >
              1.0
            </span>
          </div>
        </div>
        <div
          class="sc-bdvvtL sc-jRQBWg fGXmcv fmljMh"
        >
          <p
            class="sc-bqiRlB kMHECB"
            data-testid="route-fees"
          >
            gas-fee : ~$0.61 | ~5 min
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    id="route-6"
    style="width: 100%; margin-bottom: 1.2rem;"
  >
    <div
      aria-expanded="false"
      aria-label="select route"
      class="sc-iqseJM ibXHyJ sc-crHmcD gYEbqi"
      role="button"
      tabindex="0"
    >
      <div
        class="sc-bdvvtL sc-jRQBWg fGXmcv gzSVnD sc-egiyK ixpTqd"
      >
        <div
          class="sc-gKclnd konHGI"
        >
          <div
            class="sc-iCfMLu gYGgwA"
          >
            <svg
              class="amount__icon"
              height="3.2rem"
              name="usdcCoin"
              width="2.8rem"
            />
            <span
              class="amount__number"
            >
              1
            </span>
          </div>
          <div
            class="sc-furwcr fhgROL"
          >
            <svg
              class="arrow__icon"
              height="1.8rem"
              name="doubleArrowRight"
              width="2.1rem"
            />
          </div>
          <div
            class="sc-pVTFL joWhdh"
          >
            <svg
              class="chain__icon chain__icon--sm"
              height="2.6rem"
              name="polygonChain"
              width="2.6rem"
            />
            <p
              class="chain__name"
            >
              Polygon POS Goerli
            </p>
            <div
              class="chain__group"
            >
              <span>
                <svg
                  class="network__icon"
                  height="2.6rem"
                  name="polygonChain"
                  width="2.6rem"
                />
              </span>
              <span>
                Polygon POS Goerli
              </span>
            </div>
          </div>
          <div
            class="sc-furwcr fhgROL"
          >
            <svg
              class="arrow__icon"
              height="1.8rem"
              name="doubleArrowRight"
              width="2.1rem"
            />
          </div>
          <div
            class="sc-iCfMLu fRazBy"
          >
            <svg
              class="amount__icon"
              height="3.2rem"
              name="usdcCoin"
              width="2.8rem"
            />
            <span
              class="amount__number"
            >
              1.0
            </span>
          </div>
        </div>
        <div
          class="sc-bdvvtL sc-jRQBWg fGXmcv fmljMh"
        >
          <p
            class="sc-bqiRlB kMHECB"
            data-testid="route-fees"
          >
            gas-fee : ~$0.63 | ~30 min
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
